### PR TITLE
Add commands to list and get must-gathers from hive s3 buckets

### PR DESCRIPTION
**Pull request by fahlmant**
Creates two commands:

ocm backplane mg-log-ls # shows list of must-gathers in the hive s3 bucket
ocm backplane mg-log-extractor <s3-file-name> <clustername>-must-gather.tar.gz # extracts a must-gather from the hive s3 bucket

### CODE REVIEW

**Comment by robotmaxtron** [comment]
:+1: from me, Taylor planning some additional verification testing to follow.

**Comment by fahlmant** [comment]
/hold
Testing via the documented test process doesn't seem to work, bringing it up with the backplane team

File: scripts/SREP/mg-log-extractor/script.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  exit 10
+fi
+
+s3_log_dir="$1"
+shift # dump the specified s3_log_dir so we can pass the rest of the args to the aws command
+if [ -z "s3_log_dir" ]; then
+  echo "Error: s3_log_dir unspecified."
+  exit 10
+fi
+aws s3 sync "s3://${bucket}/${s3_log_dir}" .
```
[the remark this answers is not in the document]
Comment (fahlmant) [reply scripts/SREP/mg-log-extractor/script.sh:57]: Does the pod not get cleaned up after the managed-script runs?

[the remark this answers is not in the document]
Comment (fahlmant) [reply scripts/SREP/mg-log-extractor/script.sh:57]: Sorry, not following. What do you mean?

[the remark this answers is not in the document]
Comment (fahlmant) [reply scripts/SREP/mg-log-extractor/script.sh:57]: @georgettica can you clarify here?

[the remark this answers is not in the document]
Comment (georgettica) [reply scripts/SREP/mg-log-extractor/script.sh:57]: yeah, sorry
so what I meant is that when a job completes it'll be set to Completed and still have it's logs accessible. I am not sure if the container can be pulled down after but this data might stay there until we delete the backplane job

[the remark this answers is not in the document]
Comment (fahlmant) [reply scripts/SREP/mg-log-extractor/script.sh:57]: Can you get into a completed container?
Also other managed-scripts already merged follow this exact pattern, so this is probably a bigger problem for the framework, not something that can be solved in this PR

[the remark this answers is not in the document]
Comment (georgettica) [reply scripts/SREP/mg-log-extractor/script.sh:57]: Sure, at least I thought of this now :)
I think you can copy the file structure of an exited container

File: scripts/SREP/mg-log-extractor/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+file: script.sh
+name: mg-log-extractor
+description: |
+  Extracts must-gether logs from hive s3 bucket
```
Comment (wanghaoran1988) [inline scripts/SREP/mg-log-extractor/metadata.yaml:4]: Could you add that this script can only be ran in hive clusters?

File: scripts/SREP/mg-log-extractor/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  ocm backplane mg-log-extractor <s3-file-name> <clustername>-must-gather.tar.gz
+  tar -xvf <clustername>-must-gather.tar.gz
+author: tfahlman
+allowedGroups: 
+  - SREP
```
Comment (wanghaoran1988) [inline scripts/SREP/mg-log-extractor/metadata.yaml:9]: Do we want CEE use this script?

File: scripts/SREP/mg-log-extractor/script.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  exit 10
+fi
+
+aws s3 sync "s3://${bucket}/${s3_log_dir}" . > /dev/null 2>&1
+
+for logbundle in ./*.tar.gz; do
+  cat "$logbundle" > out.tar.gz
+done
+
+tar zcf - out.tar.gz
```
Comment (wanghaoran1988) [inline scripts/SREP/mg-log-extractor/script.sh:63]: Does this works as expected?

**Comment by wanghaoran1988** [comment]
Minor comments, others lgtm.
/approve

File: scripts/SREP/mg-log-extractor/metadata.yaml (hunk 2 above)
Comment (supreeth7) [inline scripts/SREP/mg-log-extractor/metadata.yaml:4]: Minor nits, non-blocking.
```suggestion
  Extracts must-gather logs from hive s3 bucket
```

File: scripts/SREP/mg-log-ls/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+file: script.sh
+name: mg-log-ls
+description: Lists must-gether logs in hive s3 bucket
```
Comment (supreeth7) [inline scripts/SREP/mg-log-ls/metadata.yaml:3]:
```suggestion
description: Lists must-gather logs in hive s3 bucket
```

**Review by supreeth7 — CHANGES_REQUESTED** [review]

File: scripts/SREP/mg-log-extractor/metadata.yaml (hunk 3 above)
Comment (fahlmant) [reply scripts/SREP/mg-log-extractor/metadata.yaml:9]: No. They don't have access to hives anyway I think

File: scripts/SREP/mg-log-extractor/script.sh (hunk 4 above)
Comment (fahlmant) [reply scripts/SREP/mg-log-extractor/script.sh:63]: I copied from here https://github.com/openshift/managed-scripts/blob/main/scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh#L49
as the OSD card said to. To be honest I haven't gotten this to work. Running `ocm backplane testjob logs <job>` returns a bunch of data that looks like a tar.gz but when trying to unzip it, I get errors. I've asked multiple times in the backplane slack channel but haven't had help.

Comment (wanghaoran1988) [reply scripts/SREP/mg-log-extractor/script.sh:63]: @sedroche is looking at this.

Comment (sedroche) [reply scripts/SREP/mg-log-extractor/script.sh:63]: @fahlmant I did some testing (on a simple test not your script) but pipe-ing the logs works by getting the logs from the pod directly using oc logs. The pod will be the job name in openshift-backplane-managed-scripts namespace. Can you test your script using those logs?
I think we will need some chnages in backplane to fix this

Comment (wanghaoran1988) [reply scripts/SREP/mg-log-extractor/script.sh:63]: Let's make this in as we have a workaround to use "oc logs"

**Pull request merged**